Classes/ToolRegistrator.lua

```diff
@@ -297,7 +303,17 @@ local function LeftClickToolsHook(a_Player, a_BlockX, a_BlockY, a_BlockZ, a_Bloc
 	end
 
 	local State = GetPlayerState(a_Player)
-	return State.ToolRegistrator:UseLeftClickTool(a_Player, a_BlockX, a_BlockY, a_BlockZ, a_BlockFace, a_Player:GetEquippedItem().m_ItemType)
+	State.ToolRegistrator:UseLeftClickTool(a_Player, a_BlockX, a_BlockY, a_BlockZ, a_BlockFace, a_Player:GetEquippedItem().m_ItemType)
+end
+
+
+
+
+
+local function PlayerBrokeBlockHook(a_Player, a_BlockX, a_BlockY, a_BlockZ, a_BlockFace, a_BlockType, a_BlockMeta)
+	if a_Player:GetEquippedItem().m_ItemType == E_ITEM_WOODEN_AXE then
+		a_Player:GetWorld():SetBlock(a_BlockX, a_BlockY, a_BlockZ, a_BlockType, a_BlockMeta)
+	end
```

Comment:
I don't think this will be enough for breaking multi-block structures, such as doors or beds.

Reply:
> I don't think this will be enough for breaking multi-block structures, such as doors or beds.

PLAYER_LEFT_CLICK will not be enough for breaking multi-block structures too. But thanks to PLAYER_BROKEN_BLOCK we will not have friezes.

Reply:
I don't have any freezes with the original code, so I have no idea what you're trying to fix

Reply:
But if we care about multi-block structures, should we should use PLAYER_BREAKING_BLOCK?

Reply:
> I don't have any freezes with the original code, so I have no idea what you're trying to fix

Oh, maybe i put it wrong. When you select the first point (left button of the mouse), the person stops abruptly (I think it is connected with the fact that all player actions in this packet are canceled). Friezes are not permanent, only when specifying a point

Reply:
Okay, now I see it too. When trying to select the first point while moving, the movement is stopped and the player is jerked back.

However, I think this needs a fix in Cuberite itself, rather than hacking the plugin - it seems the client doesn't like something Cuberite sends as a reaction to the rejected blockbreak.

Reply:
Well, you're right. I put everything back in place and left a bug fix when specifying the second point